arangod/Cluster/Maintenance.cpp

```diff
@@ -261,40 +271,28 @@ void handlePlanShard(VPackSlice const& cprops, VPackSlice const& ldb,
             << "for central " << dbname << "/" << colname << "- skipping";
       }
     }
-
-    // Handle leadership change, this is mostly about taking over leadership,
-    // but it also handles the case that in a failover scenario we used to
-    // be the leader and now somebody else is the leader. However, it does
-    // not handle the case of a controlled leadership resignation, see below
-    // in handleLocalShard for this.
-    if (shouldResign && !leading) {
-      // This case is a special one which is triggered if a server
-      // restarts, has `NOT_YET_TOUCHED` in its local shard as theLeader
-      // and finds a resignation sign. In that case, it should first officially
-      // take over leadership. In the following round it will then resign.
-      // This enables cleanOutServer jobs to continue to work in case of
-      // a leader restart.
-      shouldBeLeading = true;
-      shouldResign = false;
-    }
-    if ((leading != shouldBeLeading && !shouldResign) || !leaderTouched) {
+    /*
+    LOG_DEVEL_IF(dbname=="foo") << "Col: " << colname << " shard: " << shname;
+    LOG_DEVEL_IF(dbname=="foo") << std::boolalpha << "leading: " << leading << " should: " << shouldBeLeading << " resign: " << shouldResign;
+    LOG_DEVEL_IF(dbname=="foo") << "New leader: " << leaderId << " currentLeader: " << localLeader;
+    */
+    if (!leading && shouldBeLeading) {
       LOG_TOPIC("52412", DEBUG, Logger::MAINTENANCE)
           << "Triggering TakeoverShardLeadership job for shard " << dbname
           << "/" << colname << "/" << shname
           << ", local leader: " << lcol.get(THE_LEADER).copyString()
           << ", leader id: " << leaderId << ", my id: " << serverId
-          << ", should be leader: " << (shouldBeLeading ? std::string() : leaderId)
-          << ", leaderTouched = " << (leaderTouched ? "yes" : "no");
+          << ", should be leader: " << std::string();
       actions.emplace_back(ActionDescription(
-            std::map<std::string, std::string>{
-                {NAME, TAKEOVER_SHARD_LEADERSHIP},
-                {DATABASE, dbname},
-                {COLLECTION, colname},
-                {SHARD, shname},
-                {THE_LEADER, shouldBeLeading ? std::string() : leaderId},
-                {LOCAL_LEADER, std::string(localLeader)},
-                {OLD_CURRENT_COUNTER, std::to_string(feature.getCurrentCounter())}},
-                LEADER_PRIORITY));
+          std::map<std::string, std::string>{
+              {NAME, TAKEOVER_SHARD_LEADERSHIP},
+              {DATABASE, dbname},
+              {COLLECTION, colname},
+              {SHARD, shname},
+              {THE_LEADER, std::string()},
+              {LOCAL_LEADER, std::string(localLeader)},
+              {OLD_CURRENT_COUNTER, "0"}},   // legacy, no longer used
```

Comment:
```suggestion
              {OLD_CURRENT_COUNTER, std::to_string(feature.getCurrentCounter())}}, 
```
i am not too deep into the behaviors here, however in this version the Counter was filled with a sensible value.
I guess it is saver to just keep it in place.

Reply:
Done.